### PR TITLE
[CDF-2571] Close events on monitor/ruleset deletion

### PR DESCRIPTION
**Pull request by issaafalkattan**
https://cognitedata.atlassian.net/browse/CDF-2571

### CODE REVIEW

**Comment by issaafalkattan** [comment]
@mortenhillbom @hakontro

File: packages/api/src/api/models/Rulesets/Rulesets.ts [hunk 1]
```diff
@@ -262,6 +267,19 @@ class Rulesets {
       },
       parentSpan
     );
+
+    const sdk = await Projects.loginWithApiKey(
```
Comment (mortenhillbom) [inline packages/api/src/api/models/Rulesets/Rulesets.ts:271]: Do we know for 100% sure that this wont leak between projects?
Like is it/can it/should it be cleared or something between requests?

**Comment by issaafalkattan** [comment]
@mortenhillbom check again!

File: packages/api/src/api/models/Monitors/Monitors.spec.ts [hunk 2]
```diff
@@ -106,14 +111,31 @@ describe('Monitors model ', () => {
 
   it('should delete a monitor', async () => {
     await withProject(666);
+    const mockSDK = () => {
+      console.log('UGH');
```
Comment (mortenhillbom) [inline packages/api/src/api/models/Monitors/Monitors.spec.ts:115]: Remove :p

File: packages/api/src/utils/events.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    sdk: CogniteClient
+  ) => {
+    const events = await sdk.events
+      .list({
+        filter: {
+          source: DQM_EVENT_SOURCE,
+          type: DQM_EVENT_TYPE,
+          metadata: {
+            dataKitId: String(monitorId),
+            subKitId: String(ruleSetId),
```
Comment (mortenhillbom) [inline packages/api/src/utils/events.ts:18]: you can change these to `monitorId` and `ruleSetId` now :D https://github.com/cognitedata/heimdall-functions/blob/9e120ebce88a2d3e1193e0188678a1aec51717e6/src/outputHandlers/index.ts#L81

File: packages/api/src/utils/events.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    monitorId: number,
+    sdk: CogniteClient
+  ) => {
+    const events = await sdk.events
+      .list({
+        filter: {
+          source: DQM_EVENT_SOURCE,
+          type: DQM_EVENT_TYPE,
+          metadata: {
+            dataKitId: String(monitorId),
```
Comment (mortenhillbom) [inline packages/api/src/utils/events.ts:37]: same as above :)

**Pull request merged**